sql3/planner/opcopy.go

```diff
@@ -0,0 +1,515 @@
+// Copyright 2022 Molecula Corp. All rights reserved.
+
+package planner
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	pilosa "github.com/featurebasedb/featurebase/v3"
+	"github.com/featurebasedb/featurebase/v3/pql"
+	"github.com/featurebasedb/featurebase/v3/sql3"
+	"github.com/featurebasedb/featurebase/v3/sql3/parser"
+	"github.com/featurebasedb/featurebase/v3/sql3/planner/types"
+)
+
+// PlanOpCopy is a copy operator
+type PlanOpCopy struct {
+	planner     *ExecutionPlanner
+	targetTable string
+	url         string
+	apiKey      string
+	ddl         string
+	ChildOp     types.PlanOperator
+
+	warnings []string
+}
+
+func NewPlanOpCopy(planner *ExecutionPlanner, targetName string, url string, apiKey string, ddl string, child types.PlanOperator) *PlanOpCopy {
+	return &PlanOpCopy{
+		planner:     planner,
+		targetTable: targetName,
+		url:         url,
+		apiKey:      apiKey,
+		ddl:         ddl,
+		ChildOp:     child,
+		warnings:    make([]string, 0),
+	}
+}
+
+func (p *PlanOpCopy) Schema() types.Schema {
+	return types.Schema{}
+}
+
+func (p *PlanOpCopy) Iterator(ctx context.Context, row types.Row) (types.RowIterator, error) {
+	child, err := p.ChildOp.Iterator(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+	if len(p.url) > 0 {
```

Comment:
i think this is unidiomatic compared to `if p.url != ""`

Reply:
fixed